grpc-gcp/src/main/java/com/google/grpc/gcp/GcpManagedChannel.java

```diff
@@ -67,16 +86,109 @@
    *
    * @param delegateChannelBuilder the underlying delegate ManagedChannelBuilder.
    * @param apiConfig the ApiConfig object for configuring GcpManagedChannel.
+   * @param options the options for GcpManagedChannel.
    */
   public GcpManagedChannel(
-      ManagedChannelBuilder delegateChannelBuilder, ApiConfig apiConfig, int poolSize) {
+      ManagedChannelBuilder delegateChannelBuilder,
+      ApiConfig apiConfig,
+      int poolSize,
+      GcpManagedChannelOptions options) {
     loadApiConfig(apiConfig);
     if (poolSize != 0) {
       this.maxSize = poolSize;
     }
     this.delegateChannelBuilder = delegateChannelBuilder;
     // Initialize the first delegate channel.
     getChannelRef(null);
+    this.options = options;
+    initOptions();
+  }
+
+  private void initOptions() {
+    initMetrics();
+  }
+
+  private void initMetrics() {
+    final GcpMetricsOptions metricsOptions = options.getMetricsOptions();
+    if (metricsOptions == null) {
+      logger.info("Metrics options are empty. Metrics disabled.");
+      return;
+    }
+    if (metricsOptions.getMetricRegistry() == null) {
+      logger.info("Metric registry is null. Metrics disabled.");
+      return;
+    }
+    logger.info("Metrics enabled.");
+
+    final MetricRegistry metricRegistry = metricsOptions.getMetricRegistry();
+    final List<LabelKey> labelKeys = new ArrayList<>(metricsOptions.getLabelKeys());
+    final List<LabelValue> labelValues = new ArrayList<>(metricsOptions.getLabelValues());
+    labelKeys.add(LabelKey.create(POOL_INDEX_LABEL, POOL_INDEX_DESC));
+    labelValues.add(
+        LabelValue.create(String.format("pool-%d", channelPoolIndex.getAndIncrement())));
+
+    final DerivedLongGauge numChannelsMetric =
+        metricRegistry.addDerivedLongGauge(
+            metricsOptions.getNamePrefix() + METRIC_NUM_CHANNELS,
+            MetricOptions.builder()
+                .setDescription("The number of channels in the pool.")
+                .setLabelKeys(labelKeys)
+                .setUnit(GcpMetricsOptions.COUNT)
+                .build());
+
```

Comment:
creation of an object of type DerivedLongGauge is repeated below a few times.
couldn't it be done in a method so the method can be reused to create objects of type DerivedLongGauge?

Reply:
Done. Thanks!